### DISCUSSION TITLE
config 文件的注释里面含有$，可能报错

### DISCUSSION THREAD

**Discussion by hangj** [cloudwu/skynet#1073] [Ideas]
如果配置文件含有
`-- abc = "$XXX_PORT"`
skynet 加载配置时会先替换环境变量，而没有检查这个 XXX_PORT 是否在 Lua 的注释里，找不到环境变量时报错。

https://github.com/cloudwu/skynet/blob/65eb351a3b797985ab3f18c3aff6d08971b340a4/skynet-src/skynet_main.c#L103-L106

在 L104 与 L105 之间应该加一句 
`code = string.gsub(code, '%-%-.-[\\r\\n]', '\\n')\n\`
先过滤掉注释

**Comment by cloudwu** [comment]
感觉不治本。还是不混用 $ 和注释比较好。

**Comment by cloudwu** [comment]
我考虑加一个 # 注释。

**Comment by hangj** [comment]
在替换 $ 之前，string.gsub 先把注释删掉，感觉已经很治本了 :cat:

**Comment by cloudwu** [comment]
按 lua 的语法，[[ ]] 间的字符串，会忽略 — 语意。也就是说， — 未必是注释。

**Comment by hangj** [comment]
有道理
